test/OpenFeature.Contrib.ConfigCat.Test/ConfigCatProviderTest.cs

```diff
@@ -0,0 +1,101 @@
+
+            Assert.NotNull(configCatProvider.Client);
+        }
+
+        [Theory]
+        [InlineAutoData(true, false, true, ErrorType.None)]
+        [InlineAutoData(false, true, false, ErrorType.None)]
+        [InlineAutoData("false", true, true, ErrorType.TypeMismatch)]
+        public Task GetBooleanValue_ForFeature_ReturnExpectedResult(object value, bool defaultValue, bool expectedValue, ErrorType expectedErrorType, string sdkKey)
+        {
+            return ExecuteResolveTest(value, defaultValue, expectedValue, expectedErrorType, sdkKey, (provider, key, def) => provider.ResolveBooleanValue(key, def));
+        }
+
+        [Theory]
+        [InlineAutoData(1.0, 2.0, 1.0, ErrorType.None)]
+        [InlineAutoData(1, 0, 0, ErrorType.TypeMismatch)]
+        [InlineAutoData("false", 0, 0, ErrorType.TypeMismatch)]
+        [InlineAutoData(false, 0, 0, ErrorType.TypeMismatch)]
+        public Task GetDoubleValue_ForFeature_ReturnExpectedResult(object value, double defaultValue, double expectedValue, ErrorType expectedErrorType, string sdkKey)
+        {
+            return ExecuteResolveTest(value, defaultValue, expectedValue, expectedErrorType, sdkKey, (provider, key, def) => provider.ResolveDoubleValue(key, def));
+        }
+
+        [Theory]
+        [InlineAutoData("some-value", "empty", "some-value", ErrorType.None)]
+        [InlineAutoData(1, "empty", "empty", ErrorType.TypeMismatch)]
+        [InlineAutoData(false, "empty", "empty", ErrorType.TypeMismatch)]
+        public Task GetStringValue_ForFeature_ReturnExpectedResult(object value, string defaultValue, string expectedValue, ErrorType expectedErrorType, string sdkKey)
+        {
+            return ExecuteResolveTest(value, defaultValue, expectedValue, expectedErrorType, sdkKey, (provider, key, def) => provider.ResolveStringValue(key, def));
+        }
+
+        [Theory]
+        [InlineAutoData(1, 2, 1, ErrorType.None)]
+        [InlineAutoData(1.0, 0, 0, ErrorType.TypeMismatch)]
+        [InlineAutoData("false", 0, 0, ErrorType.TypeMismatch)]
+        [InlineAutoData(false, 0, 0, ErrorType.TypeMismatch)]
+        public Task GetIntValue_ForFeature_ReturnExpectedResult(object value, int defaultValue, int expectedValue, ErrorType expectedErrorType, string sdkKey)
+        {
+            return ExecuteResolveTest(value, defaultValue, expectedValue, expectedErrorType, sdkKey, (provider, key, def) => provider.ResolveIntegerValue(key, def));
+        }
+
+        [Theory]
+        [AutoData]
+        public async Task GetStructureValue_ForFeature_ReturnExpectedResult(string sdkKey)
+        {
+            const string jsonValue = "{ \"key\": \"value\" }";
+            var defaultValue = new Value(jsonValue);
+            var configCatProvider = new ConfigCatProvider(sdkKey,
+                options => { options.FlagOverrides = BuildFlagOverrides(("example-feature", defaultValue)); });

```

Comment:
Flag overrides will only work with values which can be converted to supported feature flag values (boolean, string, int, double). The ConfigCat SDK doesn't know about the `Value` type, doesn't know how to convert it to a feature flag value, so this will lead to an error.

Using a value of a supported feature flag type solves the issue:

```suggestion
                options => { options.FlagOverrides = BuildFlagOverrides(("example-feature", defaultValue.AsString)); });
```